File: backend/JobServiceMarketplace.Domain/Entities/Booking.cs
namespace JobServiceMarketplace.Domain.Entities;

public class Booking
{
    public int Id { get; set; }
    public int ClientId { get; set; }
    public User Client { get; set; } = null!;
    
    public int ProviderId { get; set; }
    public User Provider { get; set; } = null!;
    
    public string JobTitle { get; set; } = string.Empty;
    public string JobDescription { get; set; } = string.Empty;
    public DateTime ScheduledStartDate { get; set; }
    public DateTime ScheduledEndDate { get; set; }
    public decimal HourlyRate { get; set; }
    public decimal EstimatedHours { get; set; }
    public decimal TotalAmount { get; set; }
    
    public BookingStatus Status { get; set; } = BookingStatus.Pending;
    public DateTime? AcceptedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public DateTime? CancelledAt { get; set; }
    public string? CancellationReason { get; set; }
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public Payment? Payment { get; set; }
    public Review? Review { get; set; }
}

public enum BookingStatus
{
    Pending = 1,
    Accepted = 2,
    InProgress = 3,
    Completed = 4,
    Cancelled = 5,
    Disputed = 6
}

File: frontend/src/app/features/client/services/booking.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService, PaginatedResult } from '../../../core/services/api';

export interface Booking {
  id: number;
  clientId: number;
  providerId: number;
  jobTitle: string;
  jobDescription: string;
  scheduledStartDate: Date;
  scheduledEndDate: Date;
  hourlyRate: number;
  estimatedHours: number;
  totalAmount: number;
  status: number;
  acceptedAt?: Date;
  completedAt?: Date;
  createdAt: Date;
}

export interface CreateBooking {
  providerId: number;
  jobTitle: string;
  jobDescription: string;
  scheduledStartDate: Date;
  scheduledEndDate: Date;
  hourlyRate: number;
  estimatedHours: number;
}

@Injectable({
  providedIn: 'root'
})
export class BookingService {

  constructor(private apiService: ApiService) {}

  getBookings(params?: any): Observable<PaginatedResult<Booking>> {
    return this.apiService.get<PaginatedResult<Booking>>('/bookings', params);
  }

  getBooking(id: number): Observable<Booking> {
    return this.apiService.get<Booking>(`/bookings/${id}`);
  }

  createBooking(data: CreateBooking): Observable<Booking> {
    return this.apiService.post<Booking>('/bookings', data);
  }

  acceptBooking(id: number): Observable<Booking> {
    return this.apiService.post<Booking>(`/bookings/${id}/accept`, {});
  }

  completeBooking(id: number): Observable<Booking> {
    return this.apiService.post<Booking>(`/bookings/${id}/complete`, {});
  }

  cancelBooking(id: number, reason: string): Observable<Booking> {
    return this.apiService.post<Booking>(`/bookings/${id}/cancel`, { cancellationReason: reason });
  }
}
